tests/modify.spec.ts

```diff
@@ -0,0 +1,182 @@
+import { graphql } from 'graphql'
+import {
+  idArg,
+  interfaceType,
+  makeSchema,
+  NexusGraphQLSchema,
+  objectType,
+  queryField,
+  stringArg,
+} from '../src/core'
+
+describe('modify', () => {
+  let schema: NexusGraphQLSchema
+
+  beforeAll(() => {
+    schema = makeSchema({
+      types: [
+        interfaceType({
+          name: 'Node',
+          definition(t) {
+            t.id('id', {
+              description: 'Some Node ID Description',
+              resolve: () => {
+                throw new Error('Abstract')
+              },
+            })
+            t.field('subNode', {
+              type: 'Node',
+            })
+          },
+          resolveType: (o) => o.__typename,
+        }),
+        objectType({
+          name: 'User',
+          definition(t) {
+            t.implements('Node')
+            t.modify('id', {
```

Comment:
Why not just support allowing field modifications by simply declaring them. So for example here it could be `t.id(...)`.

Put another way, what's the advantage of having two different APIs for adding fields to a type versus modifying inherited fields? Instead of having one API where adding fields are treated as a modification automatically if a same-named field has been inherited?

Reply:
> Why not just support allowing field modifications by simply declaring them. So for example here it could be t.id(...).

That's how it works currently, if you define a field, you just use the existing APIs and it overwrites it. 

It's inconvenient though in situations where you have a type with `args` and you want to just change the resolver. Do you need to re-supply the args for that field, or are they merged? If they're merged, how do you know where they're coming from. 

I suppose we could make it so if the field existed already in a parent, you'd just merge them, but for `t.field`, it gets more confusing if you conditionally had to supply `type` dependent on whether or not it is inherited.

I think a `t.modify` API clearly communicates the intent, and restricts the surface area of what you can modify on a field (e.g. can't change the deprecation, or the type unless the inherited field type is an interface).

Reply:
> I think a t.modify API clearly communicates the intent, and restricts the surface area of what you can modify on a field (e.g. can't change the deprecation, or the type unless the inherited field type is an interface).

Yeah I think this is great. Both the aspect of intent clarification when reading code but also the type safety/clarity. For example I presume that `t.modify(<ctrl+tab>` would show the autocomplete for only inherited fields. That's great!